taichi/transforms/auto_diff.cpp

```diff
@@ -1115,10 +1115,32 @@ class MakeDual : public ADTransform {
       accumulate(stmt, mul(cos(stmt->operand), dual(stmt->operand)));
     } else if (stmt->op_type == UnaryOpType::cos) {
       accumulate(stmt, negate(mul(sin(stmt->operand), dual(stmt->operand))));
+    } else if (stmt->op_type == UnaryOpType::tan) {
+      TI_NOT_IMPLEMENTED
```

Comment:
OOC: why is this not implemented :P?  nit: semicolon at the end ;)

Reply:
Sure. Adding test cases (and the semicolon ).

The reason for that is the derivative of `tan (x)` is `1/ cos^2(x)`, there are many singular points for this function which causing nan. lol

Reply:
aha would you mind adding this as a comment in the code? ;)